src/resource_manager/resource_manager.py

```diff
@@ -136,11 +137,12 @@ def _name(self) -> str:
         """
 
     @abc.abstractmethod
-    def _gen_resource_from_definition(self, definition: ResourceDefinition) -> AnyResource:
+    def _gen_resource(self, definition: ResourceDefinition, config: CharmConfig) -> AnyResource:
         """Abstract method to generate a resource from ingress definition.
 
         Args:
             definition: Ingress definition to use for generating the resource.
+            config: The charm's configuration.
```

Comment:
I'm not so sure if we want to pass the whole charm config without modelling them for the module here :)

Reply:
I'm wondering if the CharmConfig class could be renamed to GatewayConfig to provide a little more suiting context for the module? :D

Reply:
That's a valid point, but the issue here is that the `resource_manager` module is an abstract class, you'll notice that the `gen_resource`  method is not implemented, so I'm passing the entire config class here since even though a `gateway` resource manager might only need the hostname and gateway class, an `httpRoute` resource manager for example might need other things (like header rewriting configuration).

Reply:
I see, thanks for the explanation!